packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1531,6 +1531,18 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             return;
         }
 
+         // If a non-local operation then go and create the object, otherwise mark it as officially attached.
+        if (this.contexts.has(attachMessage.id)) {
+            const error = new Error("DataCorruption: Duplicate data store created with existing ID");
+            this.logger.sendErrorEvent({
+                eventName: "DuplicateDataStoreId",
+                sequenceNumber: message.sequenceNumber,
+                clientId: message.clientId,
+                referenceSequenceNumber: message.referenceSequenceNumber,
+            }, error);
+            throw error;
```

Comment:
Sorry for being late, but we should not throw errors, we should close container.
throwing error assumes that there is some handler on the stack that will do the same (which is true today, but it's not the dependency that I think code like should assume).
Also it's easier to find all the reasons by we may close container if it's explicit.

See closeFn() and its usage.

Nit: It (container closure) also will automatically add sequenceNumber to logging. And I'd not log duplicate data (cllientId) - we already have that in telemetry stream, having Session_ID is enough to go up in time and see everything related to connectivity.
Also I believe I've added error type of data loss - we should fail with same error type such that it's easier to do grouping (we can sub-divide into individual cases by looking at error message).

Reply:
See ContainerErrorType.dataCorruption and its usage in delta manager.